akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -166,7 +174,10 @@ private[akka] class TLSActor(settings: ActorMaterializerSettings,
       case Some(TLSClientAuth.Need) ⇒ engine.setNeedClientAuth(true)
       case _                        ⇒ // do nothing
     }
-    params.sslParameters foreach (p ⇒ engine.setSSLParameters(p))
+    params.sslParameters foreach { p ⇒
+      hostInfo foreach { case (host, _) ⇒ applySNI(host, engine, p) }
```

Comment:
`WS` does have the same problem nowadays though, so technically it would be useful to share... on the other hand, it is just 2 lines of code so maybe not worth it...

FYI @wsargent